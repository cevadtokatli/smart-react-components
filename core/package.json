{
  "name": "@smart-react-components/core",
  "version": "2.0.0-beta-0",
  "license": "MIT",
  "main": "index.js",
  "description": "SRC core includes the theme, Styled properties, and Element components.",
  "keywords": [
    "javascript",
    "typescript",
    "react",
    "styled-components"
  ],
  "author": {
    "name": "Cevad Tokatlı",
    "email": "cevadtokatli@hotmail.com"
  },
  "contributors": [
    {
      "name": "Cemil Tokatlı",
      "email": "cemil@tokatli.me"
    }
  ],
  "repository": {
    "type": "git",
    "url": "https://github.com/cevadtokatli/smart-react-components/tree/v2/core"
  },
  "publishConfig": {
    "access": "public"
  },
  "scripts": {
    "build": "yarn clear && tsc --outDir lib && yarn copy",
    "clear": "rm -rf lib",
    "copy": "cp package.json lib/package.json && cp README.md lib/README.md && cp ../LICENSE lib/LICENSE",
    "dev": "tsc -w --outDir lib",
    "lint": "eslint src",
    "release": "../scripts/release.sh core",
    "test": "jest"
  },
  "devDependencies": {
    "@types/color": "3.0.3",
    "@types/enzyme": "3.10.12",
    "@types/jest": "29.0.1",
    "@types/react": "18.0.19",
    "@types/react-dom": "18.0.6",
    "@types/styled-components": "5.1.26",
    "@wojtekmaj/enzyme-adapter-react-17": "0.6.7",
    "enzyme": "3.11.0",
    "eslint": "8.0.1",
    "jest": "29.0.3",
    "jsdom": "20.0.0",
    "jsdom-global": "3.0.2",
    "react": "18.2.0",
    "react-dom": "18.2.0",
    "styled-components": "5.3.5",
    "ts-jest": "29.0.0",
    "typescript": "4.8.3"
  },
  "dependencies": {
    "color": "4.2.3"
  },
  "peerDependencies": {
    "react": "^18.2.0",
    "styled-components": "^5.3.5"
  }
}
