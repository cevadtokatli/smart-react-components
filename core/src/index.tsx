export * from './theme'
export { default as createTheme } from './theme'
export * from './styled-props'
export * from './Element'
