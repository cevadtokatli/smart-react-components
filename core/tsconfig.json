{
  "extends": "../tsconfig.json",
  "include": ["src/**/*.tsx"],
  "exclude": ["node_modules", "src/**/*.test.tsx"]
}
