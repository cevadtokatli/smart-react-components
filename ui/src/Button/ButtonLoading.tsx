import { PaletteProp } from '@smart-react-components/core/types'
import CSSTransition from '@smart-react-components/transition/CSSTransition'
import React from 'react'
import ButtonLoadingElement from '../components/Button/ButtonLoadingElement'
import Loading from '../Loading'

interface Props {
  children?: JSX.Element
  hasAnimation?: boolean
}

interface PrivateProps {
  isOutline: boolean
  isSoft: boolean
  palette: PaletteProp
  status: boolean
}

const ButtonLoading: React.FC<Props> = ({ children, hasAnimation = true, isOutline, isSoft, palette, status }: Props & PrivateProps) => {
  const getColor = () => `!${palette}.${
    !isSoft
      ? !isOutline ? 'font' : 'main'
      : !isOutline ? 'softFont' : 'soft'
  }`

  return (
    <CSSTransition className="src-button-loading" duration={300} hasHideAnimation={hasAnimation} hasShowAnimation={hasAnimation} status={status}>
      <ButtonLoadingElement hasAnimation={hasAnimation}>
        { children ?? <Loading color={getColor()} /> }
      </ButtonLoadingElement>
    </CSSTransition>
  )
}

ButtonLoading.displayName = 'SRCLoading'

export default ButtonLoading
