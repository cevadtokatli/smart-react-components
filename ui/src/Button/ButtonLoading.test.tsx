import { wrapTheme } from '@smart-react-components/core/test'
import { render } from '@testing-library/react'
import React from 'react'
import ButtonLoading from './ButtonLoading'

describe('<ButtonLoading />', () => {
  let props

  beforeAll(() => {
    props = {
      hasAnimation: false,
      isOutline: false,
      isSoft: false,
      palette: 'primary',
      status: true,
    }
  })

  it('should render component', () => {
    const { asFragment } = render(wrapTheme(<ButtonLoading {...props} />))
    expect(asFragment()).toMatchSnapshot()
  })

  it('should render component with custom content', () => {
    const { asFragment } = render(wrapTheme(<ButtonLoading {...props}><div /></ButtonLoading>))
    expect(asFragment()).toMatchSnapshot()
  })
})
