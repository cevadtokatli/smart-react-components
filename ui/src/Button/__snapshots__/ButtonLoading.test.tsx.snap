// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ButtonLoading /> should render component 1`] = `
<DocumentFragment>
  .c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  left: 0;
  position: absolute;
  width: 100%;
  z-index: 10;
}

.c3 {
  stroke: #fff;
  stroke-width: 4px;
}

.c1 {
  height: 20px;
  width: 20px;
}

.c2 {
  -webkit-animation: src-loading-animation 2000ms linear infinite;
  animation: src-loading-animation 2000ms linear infinite;
}

.c2 circle {
  -webkit-animation: src-loading-circle-animation 2000ms linear infinite;
  animation: src-loading-circle-animation 2000ms linear infinite;
  fill: none;
  stroke-dasharray: 1,150;
  stroke-dashoffset: 0;
  stroke-linecap: round;
}

<div
    class=" c0"
  >
    <svg
      class="c1 c2"
      viewBox="0 0 44 44"
    >
      <circle
        class="c3"
        cx="22"
        cy="22"
        r="20"
        stroke="!primary.font"
        stroke-width="4"
      />
    </svg>
  </div>
</DocumentFragment>
`;

exports[`<ButtonLoading /> should render component with custom content 1`] = `
<DocumentFragment>
  .c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  left: 0;
  position: absolute;
  width: 100%;
  z-index: 10;
}

<div
    class=" c0"
  >
    <div />
  </div>
</DocumentFragment>
`;
